### PR TITLE
fix(core): ignore hidden files when loading skill resources

### LINKED ISSUE #1000: [Bug]:skill resource should filter hidden file
Image

### PR DESCRIPTION
**Pull request by jujn**
## Description
Close #1000

## Checklist

Please check the following items before code is ready to be reviewed.

- [ ]  Code has been formatted with `mvn spotless:apply`
- [ ]  All tests are passing (`mvn test`)
- [ ]  Javadoc comments are complete and follow project conventions
- [ ]  Related documentation has been updated (e.g. links, examples, etc.)
- [ ]  Code is ready for review

### CODE REVIEW

**Comment by LearningGp** [comment]
@fang-tech PTAL

**Review by LearningGp — APPROVED** [review]
LGTM

File: agentscope-core/src/test/java/io/agentscope/core/skill/util/SkillFileSystemHelperTest.java [hunk 1]
```diff
@@ -255,6 +258,116 @@ void testSaveSkills_DecodesBase64ToBinary() throws IOException {
         assertEquals("plain text", Files.readString(savedText, StandardCharsets.UTF_8));
     }
 
+    @Test
+    @DisplayName(
+            "Should fully cover resource filtering logic (unreadable, dot-files, OS hidden, and"
+                    + " IOException)")
+    void testLoadResources_FiltersAllEdgeCases() throws IOException {
```
Comment (fang-tech) [inline agentscope-core/src/test/java/io/agentscope/core/skill/util/SkillFileSystemHelperTest.java]: Split your tests into individual unit tests for each case.

File: agentscope-core/src/test/java/io/agentscope/core/skill/util/SkillFileSystemHelperTest.java [hunk 2]
```diff
@@ -255,6 +258,116 @@ void testSaveSkills_DecodesBase64ToBinary() throws IOException {
...
+        // hide files in the folder
+        Path dotDir = skillDir.resolve(".hidden_dir");
+        Files.createDirectories(dotDir);
+        Path dotDirFile = dotDir.resolve("config.txt");
+        Files.writeString(dotDirFile, "hidden config", StandardCharsets.UTF_8);
+
+        // hidden files at the operating system level
+        Path osHiddenFile = skillDir.resolve("os_hidden_file.txt");
+        Files.writeString(osHiddenFile, "hidden data", StandardCharsets.UTF_8);
+        boolean isWindows = System.getProperty("os.name").toLowerCase().contains("win");
```
Comment (fang-tech) [inline agentscope-core/src/test/java/io/agentscope/core/skill/util/SkillFileSystemHelperTest.java]: For OS environment-specific testing, use `@EnabledOnOs(OS.WINDOWS)` annotation

File: agentscope-core/src/main/java/io/agentscope/core/skill/util/SkillFileSystemHelper.java [hunk 3]
```diff
@@ -359,6 +359,23 @@ private static Map<String, String> loadResources(Path skillDir, Path skillFile)
         try (Stream<Path> paths = Files.walk(skillDir)) {
```
Comment (fang-tech) [inline agentscope-core/src/main/java/io/agentscope/core/skill/util/SkillFileSystemHelper.java]: For the dot-hidden directories, we should directly filter them out instead of walking through them and then filtering again.

File: agentscope-core/src/main/java/io/agentscope/core/skill/util/SkillFileSystemHelper.java [hunk 4]
```diff
@@ -359,6 +359,23 @@ private static Map<String, String> loadResources(Path skillDir, Path skillFile)
...
             paths.filter(Files::isRegularFile)
                     .filter(p -> !p.equals(skillFile))
+                    .filter(
+                            p -> {
```
Comment (fang-tech) [inline agentscope-core/src/main/java/io/agentscope/core/skill/util/SkillFileSystemHelper.java]: It would be more readable to extract this part into a separate private method.

File: agentscope-core/src/test/java/io/agentscope/core/skill/util/SkillFileSystemHelperTest.java [hunk 5]
```diff
@@ -255,6 +258,116 @@ void testSaveSkills_DecodesBase64ToBinary() throws IOException {
         assertEquals("plain text", Files.readString(savedText, StandardCharsets.UTF_8));
     }
 
+    @Test
+    @DisplayName(
```
Comment (fang-tech) [inline agentscope-core/src/test/java/io/agentscope/core/skill/util/SkillFileSystemHelperTest.java]: Split each edge case into a separate test method

File: agentscope-core/src/test/java/io/agentscope/core/skill/util/SkillFileSystemHelperTest.java (hunk 1 above)
Comment (jujn) [reply agentscope-core/src/test/java/io/agentscope/core/skill/util/SkillFileSystemHelperTest.java]: Thanks for the suggestion! Considering that the business logic of resource filtering in this section is actually very lightweight. I understand that splitting test cases can make responsibilities more singular, but it can make the test classes cumbersome. Hope you can reconsider?

Comment (fang-tech) [reply agentscope-core/src/test/java/io/agentscope/core/skill/util/SkillFileSystemHelperTest.java]: If the code of the filter needs to be modified in the future, things will become complicated. We won't be able to precisely determine which aspects of the tests were made unavailable due to our modifications. You might think it's too cumbersome and could appropriately combine them. However, if all the tests are placed in one method, it will impose a burden on future modifications. This is actually a formatting task, and you can use AI to complete it quickly.

Comment (jujn) [reply agentscope-core/src/test/java/io/agentscope/core/skill/util/SkillFileSystemHelperTest.java]: OK. Done
